### ISSUE TITLE
fix useChain

### ISSUE THREAD

**Issue by Sen-Com** [cosmology-tech/cosmos-kit#361]
useChain is not correctly picking up all events.

Consider

> ....
> const {address} = useChain("cosmosHub")
> return <div>{address}</div>
> ....

If I connect and change accounts, it works. 
If I then re-connect and change accounts it does not pick up any account changes, i.e. I change account and the address does not change.

The bug is also in production, i.e. Osmosis frontend.
even forcefully refreshing the component a la
>  useEffect(() =>  () => setRefreshKey(refreshKey + 1), [chain, chainName])

is not working

**Comment by Sen-Com** [comment]
local storage items are also not correct updated (i.e. no change) on account change

**Comment by Sen-Com** [comment]
interestingly, useChainWallet does not suffer from the same issue and always picks up account changes correctly. That's a clear indication that the useChain hook is broken

**Comment by liujun93** [comment]
@Sen-Com What did you do to cause `re-connect`? Usually `Reconnect` button shows up when you failed to connect, i.e. connection rejected by wallet. I tries this and reconnect, switch account still works. Is this not the `re-connect` you mentioned? can you list the exact steps so that we can reproduce the issue?

**Comment by Sen-Com** [comment]
I tried countless things to make this work and all lead to the same issue: the useChain hook is not registering account changes correctly.

1. connect wallet
2. switch account -> registered
3. disconnect wallet
4. connect wallet
5. switch account -> not registered

**Comment by liujun93** [comment]
I still cannot reproduce it. Are you using the latest version of CosmosKit?

Maybe you can also try these steps in our doc site. https://docs.cosmoskit.com/#try-it-out

I'm using Keplr extension to test and everything works fine.

Or, can you make a simple example repo to reproduce it?

**Comment by bongdungyeuem27** [comment]
> I tried countless things to make this work and all lead to the same issue: the useChain hook is not registering account changes correctly.
> 
> 1. connect wallet
> 2. switch account -> registered
> 3. disconnect wallet
> 4. connect wallet
> 5. switch account -> not registered

Yes, I have the same issue with nextjs page. And It even can not track when i use dynamic import
